support/modules-artifact-gen/single-file-artifact-gen

```diff
@@ -156,7 +156,11 @@ echo "$dest_dir" > $dest_dir_file
 echo "$filename" > $filename_file
 
 # Create permissions file in plain text
-stat -c %a "${file}" > $permissions_file
+if [[ "$OSTYPE" == "darwin"* ]]; then
```

Comment:
I didn't know about this env var. I would have suggested using `uname -c`

Reply:
from what I have seen, uname might produce differently capitalized results while the env var should be at least somewhat reliable - not entirely sure though